File: lib/session.js
const axios = require('axios');
const { logger } = require('./log');
const config = require('../config.json');

let session;

async function auth() {
  try {
    const authRes = await axios({
      method: 'post',
      url: '/auth',
      baseURL: config.mirai.apiBaseURL,
      data: {
        authKey: config.mirai.authKey,
      },
    });
    const { data: authData } = authRes;
    if (authData.code === 0) {
      session = authData.session;
      logger.info('Auth succeed.');
      logger.debug(authData);
    } else {
      logger.fatal(`Auth failed with code ${authData.code}.`);
      logger.debug(authData);
      process.exit(-1);
    }
  } catch (error) {
    logger.fatal(`Auth failed. ${error.message}`);
    logger.debug(error);
    process.exit(-1);
  }
}

async function verify() {
  try {
    const verifyRes = await axios({
      method: 'post',
      url: '/verify',
      baseURL: config.mirai.apiBaseURL,
      data: {
        sessionKey: session,
        qq: config.mirai.qq,
      },
    });
    const { data: verifyData } = verifyRes;
    if (verifyData.code === 0) {
      logger.info('Verify session succeed.');
      logger.debug(verifyData);
    } else {
      logger.fatal(`Verify session failed with code ${verifyData.code}.`);
      logger.debug(verifyData);
      process.exit(-1);
    }
  } catch (error) {
    logger.fatal(`Verify session failed. ${error.message}`);
    logger.debug(error);
    process.exit(-1);
  }
}

async function initSession() {
  await auth();
  await verify();
}

function getSession() {
  return session;
}

module.exports = {
  initSession,
  getSession,
};

File: session.go
package main

import (
	"github.com/go-resty/resty/v2"
	"github.com/rs/zerolog/log"
	"github.com/tidwall/gjson"
)

var session string

func initSession() {
	auth()
	verify()
}

func auth() {
	client := resty.New()
	client.SetCloseConnection(true)
	resp, err := client.R().
		SetHeader("Content-Type", "application/json").
		SetBody(map[string]interface{}{
			"authKey": cfg.MustValue("mirai", "authKey"),
		}).
		Post(cfg.MustValue("mirai", "apiBaseURL") + "/auth")
	if err != nil {
		log.Fatal().Msgf("Auth failed. %v", err)
	}
	code := gjson.GetBytes(resp.Body(), "code")
	if code.Exists() && code.Value() == 0.0 {
		log.Info().Msg("Auth succeed.")
		log.Debug().Msgf("%v", resp)
		session = gjson.GetBytes(resp.Body(), "session").String()
		return
	}
	log.Debug().Msgf("%v", resp)
	log.Fatal().Msg("Auth failed.")
}

func verify() {
	client := resty.New()
	client.SetCloseConnection(true)
	resp, err := client.R().
		SetHeader("Content-Type", "application/json").
		SetBody(map[string]interface{}{
			"sessionKey": session,
			"qq":         cfg.MustValue("mirai", "qq"),
		}).
		Post(cfg.MustValue("mirai", "apiBaseURL") + "/verify")
	if err != nil {
		log.Fatal().Msgf("Verify failed. %v", err)
	}
	code := gjson.GetBytes(resp.Body(), "code")
	if code.Exists() && code.Value() == 0.0 {
		log.Info().Msg("Verify succeed.")
		log.Debug().Msgf("%v", resp)
		return
	}
	log.Debug().Msgf("%v", resp)
	log.Fatal().Msg("Verify failed.")
}
